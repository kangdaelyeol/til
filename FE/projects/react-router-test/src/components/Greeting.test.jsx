import { render } from '@testing-library/react'
import Greeting from './Greeting'
import { expect } from 'vitest'

describe('Greeting Component test', () => {
    test('givenNameAsAliceWhenRendersComponentThenWorksWell', () => {
        const { getByText } = render(<Greeting name={'Alice'} />)
        expect(getByText('Hello, Alice')).toBeInTheDocument()
    })

    it('should render', () => {
        const { asFragment } = render(<Greeting name={'Alice'} />)

        expect(asFragment()).toMatchSnapshot()
    })
})
