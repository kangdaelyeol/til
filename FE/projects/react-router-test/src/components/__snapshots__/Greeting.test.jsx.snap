// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Greeting Component test > should render 1`] = `
<DocumentFragment>
  <div>
    Hello, Alice
  </div>
</DocumentFragment>
`;
